### ISSUE TITLE
unz<func>() not present in compat mode

### ISSUE THREAD

**Issue by MilhouseVH** [zlib-ng/zlib-ng#486]
Thinking about it, this may be related to #482. I've opened a new issue but feel free to close it and continue with #482/#485 if that's the best way to solve this.

The `unzClose()`, `unzOpen()` (etc.) functions don't seem to be present in `zlib-ng` with `--zlib-compat`, or perhaps the `OF()` macro implemented in #485 is not working as it should.

I'm testing with `--zlib-compat` and #485 (which implements the `OF()` macro).

`tntnet` ([website](http://www.tntnet.org/)) and `libretro-uae4arm` ([website](https://github.com/libretro/uae4arm-libretro)) have stopped complaining about the missing `OF()` macro, but now they're failing as follows:

### `tntnet`
```
/home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzipfile.cpp:110:46: error: too many arguments to function 'void* unzOpen()'
     if (!(file->file = ::unzOpen(path.c_str())))
                                              ^
In file included from /home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzipfile.cpp:32:0:
/home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzip.h:166:24: note: declared here
 extern unzFile ZEXPORT unzOpen OF((const char *path));
                        ^
/home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzipfile.cpp: In member function 'void tnt::unzipFile::close()':
/home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzipfile.cpp:122:26: error: too many arguments to function 'int unzClose()'
       unzClose(file->file);
                          ^
In file included from /home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzipfile.cpp:32:0:
/home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzip.h:198:20: note: declared here
 extern int ZEXPORT unzClose OF((unzFile file));
                    ^
/home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzipfile.cpp: In destructor 'tnt::unzipFile::~unzipFile()':
/home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzipfile.cpp:132:26: error: too many arguments to function 'int unzClose()'
       unzClose(file->file);
                          ^
In file included from /home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzipfile.cpp:32:0:
/home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzip.h:198:20: note: declared here
 extern int ZEXPORT unzClose OF((unzFile file));
                    ^
/home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzipfile.cpp: In member function 'void tnt::unzipFile::goToFirstFile()':
/home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzipfile.cpp:139:45: error: too many arguments to function 'int unzGoToFirstFile()'
     checkError(::unzGoToFirstFile(file->file), "unzGoToFirstFile");
                                             ^
In file included from /home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzipfile.cpp:32:0:
/home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzip.h:229:20: note: declared here
 extern int ZEXPORT unzGoToFirstFile OF((unzFile file));
                    ^
/home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzipfile.cpp: In member function 'void tnt::unzipFile::goToNextFile()':
/home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzipfile.cpp:144:44: error: too many arguments to function 'int unzGoToNextFile()'
     checkError(::unzGoToNextFile(file->file), "unzGoToNextFile");
                                            ^
In file included from /home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzipfile.cpp:32:0:
/home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzip.h:235:20: note: declared here
 extern int ZEXPORT unzGoToNextFile OF((unzFile file));
                    ^
/home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzipfile.cpp: In member function 'void tnt::unzipFile::locateFile(const string&, bool)':
/home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzipfile.cpp:149:85: error: too many arguments to function 'int unzLocateFile()'
     checkError(::unzLocateFile(file->file, fileName.c_str(), caseSensitivity ? 1 : 0), "unzLocateFile");
                                                                                     ^
In file included from /home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzipfile.cpp:32:0:
/home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzip.h:242:20: note: declared here
 extern int ZEXPORT unzLocateFile OF((unzFile file,
                    ^
/home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzipfile.cpp: In member function 'void tnt::unzipFile::openCurrentFile()':
/home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzipfile.cpp:154:47: error: too many arguments to function 'int unzOpenCurrentFile()'
     checkError(::unzOpenCurrentFile(file->file), "unzOpenCurrentFile");
                                               ^
In file included from /home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzipfile.cpp:32:0:
/home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzip.h:331:20: note: declared here
 extern int ZEXPORT unzOpenCurrentFile OF((unzFile file));
                    ^
/home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzipfile.cpp: In member function 'void tnt::unzipFile::openCurrentFile(const string&)':
/home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzipfile.cpp:159:67: error: too many arguments to function 'int unzOpenCurrentFilePassword()'
     checkError(::unzOpenCurrentFilePassword(file->file, pw.c_str()), "unzOpenCurrentFilePassword");
                                                                   ^
In file included from /home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzipfile.cpp:32:0:
/home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzip.h:337:20: note: declared here
 extern int ZEXPORT unzOpenCurrentFilePassword OF((unzFile file,
                    ^
/home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzipfile.cpp: In member function 'void tnt::unzipFile::closeCurrentFile()':
/home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzipfile.cpp:164:48: error: too many arguments to function 'int unzCloseCurrentFile()'
     checkError(::unzCloseCurrentFile(file->file), "unzCloseCurrentFile");
                                                ^
In file included from /home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzipfile.cpp:32:0:
/home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzip.h:373:20: note: declared here
 extern int ZEXPORT unzCloseCurrentFile OF((unzFile file));
                    ^
/home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzipfile.cpp: In member function 'int tnt::unzipFile::readCurrentFile(void*, unsigned int)':
/home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzipfile.cpp:169:64: error: too many arguments to function 'int unzReadCurrentFile()'
     return checkError(::unzReadCurrentFile(file->file, buf, len), "unzReadCurrentFile");
                                                                ^
In file included from /home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzipfile.cpp:32:0:
/home/ubuntu/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/tntnet-2.2.1/framework/common/unzip.h:379:20: note: declared here
 extern int ZEXPORT unzReadCurrentFile OF((unzFile file,
                    ^
make[2]: *** [Makefile:871: libtntnet_la-unzipfile.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
```
`tntnet` build log: http://ix.io/22Sn

### `libretro-uae4arm`
```
src/zfile_archive.cpp: In function 'void archive_close_zip(void*)':
src/zfile_archive.cpp:203:20: error: too many arguments to function 'int unzClose()'
  203 |     unzClose(handle);
      |                    ^
In file included from src/zfile_archive.cpp:14:
src/archivers/zip/unzip.h:147:20: note: declared here
  147 | extern int ZEXPORT unzClose OF((unzFile file));
      |                    ^~~~~~~~
src/zfile_archive.cpp: In function 'zvolume* archive_directory_zip(zfile*)':
src/zfile_archive.cpp:214:20: error: too many arguments to function 'void* unzOpen()'
  214 |     uz = unzOpen (z);
      |                    ^
In file included from src/zfile_archive.cpp:14:
src/archivers/zip/unzip.h:136:24: note: declared here
  136 | extern unzFile ZEXPORT unzOpen OF((struct zfile *path));
      |                        ^~~~~~~
src/zfile_archive.cpp:217:29: error: too many arguments to function 'int unzGoToFirstFile()'
  217 |     if (unzGoToFirstFile (uz) != UNZ_OK)
      |                             ^
In file included from src/zfile_archive.cpp:14:
src/archivers/zip/unzip.h:175:20: note: declared here
  175 | extern int ZEXPORT unzGoToFirstFile OF((unzFile file));
      |                    ^~~~~~~~~~~~~~~~
src/zfile_archive.cpp:225:102: error: too many arguments to function 'int unzGetCurrentFileInfo()'
  225 |  err = unzGetCurrentFileInfo(uz, &file_info, filename_inzip, sizeof(filename_inzip), NULL, 0, NULL, 0);
      |                                                                                                      ^
In file included from src/zfile_archive.cpp:14:
src/archivers/zip/unzip.h:201:20: note: declared here
  201 | extern int ZEXPORT unzGetCurrentFileInfo OF((unzFile file,
      |                    ^~~~~~~~~~~~~~~~~~~~~
src/zfile_archive.cpp:235:38: error: too many arguments to function 'int unzOpenCurrentFile()'
  235 |      int err = unzOpenCurrentFile (uz);
      |                                      ^
In file included from src/zfile_archive.cpp:14:
src/archivers/zip/unzip.h:227:20: note: declared here
  227 | extern int ZEXPORT unzOpenCurrentFile OF((unzFile file));
      |                    ^~~~~~~~~~~~~~~~~~
src/zfile_archive.cpp:245:27: error: too many arguments to function 'int unzGoToNextFile()'
  245 |  err = unzGoToNextFile (uz);
      |                           ^
In file included from src/zfile_archive.cpp:14:
src/archivers/zip/unzip.h:181:20: note: declared here
  181 | extern int ZEXPORT unzGoToNextFile OF((unzFile file));
      |                    ^~~~~~~~~~~~~~~
src/zfile_archive.cpp: In function 'zfile* archive_access_zip(znode*)':
src/zfile_archive.cpp:262:29: error: too many arguments to function 'int unzGoToFirstFile()'
  262 |     if (unzGoToFirstFile (uz) != UNZ_OK)
      |                             ^
In file included from src/zfile_archive.cpp:14:
src/archivers/zip/unzip.h:175:20: note: declared here
  175 | extern int ZEXPORT unzGoToFirstFile OF((unzFile file));
      |                    ^~~~~~~~~~~~~~~~
src/zfile_archive.cpp:268:34: error: too many arguments to function 'int unzLocateFile()'
  268 |     if (unzLocateFile (uz, tmp, 1) != UNZ_OK) {
      |                                  ^
In file included from src/zfile_archive.cpp:14:
src/archivers/zip/unzip.h:188:20: note: declared here
  188 | extern int ZEXPORT unzLocateFile OF((unzFile file,
      |                    ^~~~~~~~~~~~~
src/zfile_archive.cpp:273:31: error: too many arguments to function 'int unzLocateFile()'
  273 |  if (unzLocateFile (uz, tmp, 1) != UNZ_OK)
      |                               ^
In file included from src/zfile_archive.cpp:14:
src/archivers/zip/unzip.h:188:20: note: declared here
  188 | extern int ZEXPORT unzLocateFile OF((unzFile file,
      |                    ^~~~~~~~~~~~~
src/zfile_archive.cpp:276:31: error: too many arguments to function 'int unzOpenCurrentFile()'
  276 |     if (unzOpenCurrentFile (uz) != UNZ_OK)
      |                               ^
In file included from src/zfile_archive.cpp:14:
src/archivers/zip/unzip.h:227:20: note: declared here
  227 | extern int ZEXPORT unzOpenCurrentFile OF((unzFile file));
      |                    ^~~~~~~~~~~~~~~~~~
src/zfile_archive.cpp:280:49: error: too many arguments to function 'int unzReadCurrentFile()'
  280 |  err = unzReadCurrentFile (uz, z->data, zn->size);
      |                                                 ^
In file included from src/zfile_archive.cpp:14:
src/archivers/zip/unzip.h:240:20: note: declared here
  240 | extern int ZEXPORT unzReadCurrentFile OF((unzFile file,
      |                    ^~~~~~~~~~~~~~~~~~
src/zfile_archive.cpp:282:28: error: too many arguments to function 'int unzCloseCurrentFile()'
  282 |     unzCloseCurrentFile (uz);
      |                            ^
```
`libretro-uae4arm` build log: http://ix.io/22Sf

**Comment by mtl1979** [comment]
I think I need to short-circuit the test in the OF patch... it's taking the wrong variation...

**Comment by MilhouseVH** [comment]
Closing, fixed in #485.
